### ISSUE TITLE
"Find references" doesn't recognize `.type` usages

### ISSUE THREAD

**Issue by kubukoz** [scalameta/metals#1633]
**Describe the bug**
"Find all references" misses usages of an object in references to its singleton type.

**To Reproduce**
1. Create a file with this content:

```scala
object root {

  object impl

  //the following two usages of impl aren't found
  val f: impl.type => Unit = {
    case _: impl.type => ()
  }

  //this one is found
  f(impl)

}

```

2. Save it, compile, look for references of `impl`.

3. Only the last reference is found

**Expected behavior**
All 3 references are found

**Screenshots**

(peek references)
![image](https://user-images.githubusercontent.com/894884/79519964-b4ac7d00-8055-11ea-8595-14aa256d5ca2.png)

**Installation:**
- Operating system: macOS
- Editor: Visual Studio Code
- Metals version: v0.8.4

**Extra information**

The `case` keyword on the object has no effect on this behavior.

**Comment by kubukoz** [comment]
Also, when the cursor is positioned like this:

![image](https://user-images.githubusercontent.com/894884/79520016-d9085980-8055-11ea-9cc0-84151c4ca5b8.png)

the definition isn't found, even though the name appears. Ideally, you would be able to find the definition if you were at any character of the `impl.type` string.

**Comment by tgodzik** [comment]
Thanks for reporting, looks like an issue inside scalameta's semanticDB.

**Comment by tgodzik** [comment]
This is fixed for 2.13.5 and and will be fixed for 2.12.14. It needed a fix in the compiler itself. Thanks @tanishiking !

**Comment by kubukoz** [comment]
😍 😍 😍 😍 😍
